libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/DefaultTracer.java

```diff
@@ -36,41 +36,11 @@ public DefaultTracer(TracingTelemetry tracingTelemetry, TracerContextStorage<Str
     }
 
     @Override
-    public Scope startSpan(String spanName) {
+    public SpanScope startSpan(String spanName) {
         Span span = createSpan(spanName, getCurrentSpan());
         setCurrentSpanInContext(span);
```

Comment:
With all the methods exposed in SpanScope, we expect to have scope object explicitly and don't expect magic through threadContext? Do we still need to set it in threadContext?

Reply:
Yes. We need thread context to save the current active span to be used as parent for any new span